packages/vm/src/calls.rs

```diff
@@ -734,6 +738,86 @@ mod tests {
         assert_eq!(query_response.as_slice(), b"{\"verifier\":\"verifies\"}");
     }
 
+    #[test]
+    fn float_instrs_are_deterministic() {
+        #[derive(Debug, serde::Serialize, serde::Deserialize)]
+        #[serde(rename_all = "snake_case")]
+        pub enum Value {
+            U32(u32),
+            U64(u64),
+            F32(u32),
+            F64(u64),
+        }
+
+        let mut instance = mock_instance_with_options(
+            FLOATY2,
+            MockInstanceOptions {
+                gas_limit: u64::MAX,
+                memory_limit: None,
+                ..Default::default()
+            },
+        );
+
+        // init
+        let info = mock_info("creator", &[]);
+        call_instantiate::<_, _, _, Empty>(&mut instance, &mock_env(), &info, br#"{}"#)
+            .unwrap()
+            .unwrap();
+
+        // query instructions
+        let msg = br#"{"instructions":{}}"#;
+        let contract_result = call_query(&mut instance, &mock_env(), msg)
+            .unwrap()
+            .unwrap();
+        let instructions: Vec<String> = from_json(&contract_result).unwrap();
+        // little sanity check
+        assert_eq!(instructions.len(), 70);
+
+        const RUNS_PER_INSTRUCTION: u64 = 150;
+        let mut hasher = Sha256::new();
+        for instr in &instructions {
+            for seed in 0..RUNS_PER_INSTRUCTION {
+                // query some input values for the instruction
+                let args: Vec<Value> = from_json(
+                    &call_query(
+                        &mut instance,
+                        &mock_env(),
+                        format!(
+                            r#"{{"random_args_for":{{ "instruction": "{instr}", "seed": {seed}}}}}"#
+                        )
+                        .as_bytes(),
+                    )
+                    .unwrap()
+                    .unwrap(),
+                )
+                .unwrap();
+
+                // build the run message
+                let args = to_json_string(&args).unwrap();
+                let msg: String = format!(
+                    r#"{{"run":{{
+                        "instruction": "{instr}",
+                        "args": {args}
+                    }}}}"#
+                );
+                // run the instruction
+                // this might throw a runtime error (e.g. if the instruction traps)
+                let result = match call_query(&mut instance, &mock_env(), msg.as_bytes()) {
+                    Ok(ContractResult::Ok(r)) => format!("{:?}", from_json::<Value>(&r).unwrap()),
+                    Err(VmError::RuntimeErr { msg }) => msg,
```

Comment:
Shouldn't this look like:
```Err(VmError::RuntimeErr { msg, .. }) => msg,`

Reply:
Ok, my remark is valid only when **backtraces** feature is switched on. It's ok, until tests are not run with this feature.

Reply:
That's true, but still a good catch.